app/models/pet.rb

```diff
@@ -10,4 +12,16 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search(name)
```

Comment:
Nice work encapsulating this at the model level, but it doesn't seem like you used this in your controller. Also, check out ApplicationRecord -- it's already written for you!